src/main/scala/com/johnsnowlabs/nlp/annotators/sbd/deep/DeepSentenceDetector.scala

```diff
@@ -55,20 +62,32 @@ class DeepSentenceDetector(override val uid: String) extends AnnotatorModel[Deep
 
   }
 
+  override protected def afterAnnotate(dataset: DataFrame): DataFrame = {
+    import org.apache.spark.sql.functions.{col, explode, array}
+    if ($(explodeSentences)) {
+      dataset
+        .select(dataset.columns.filterNot(_ == getOutputCol).map(col) :+ explode(col(getOutputCol)).as("_tmp"):_*)
+        .withColumn(getOutputCol, array(col("_tmp"))
+          .as(getOutputCol, dataset.schema.fields.find(_.name == getOutputCol).get.metadata))
+        .drop("_tmp")
+    }
+    else dataset
+  }
+
   def getDocument(annotations: Seq[Annotation]): Seq[Annotation] = {
-    annotations.filter(annotation => annotation.annotatorType == DOCUMENT)
+    Seq(annotations.find(annotation => annotation.annotatorType == DOCUMENT).head)
```

Comment:
Agreed, the code was modified to take that into account.

Reply:
This is interesting. Since SentenceDetector receives something not "sentence detected", it looks "safe" to assume there is only going to be one DOCUMENT. This also verifies that it is not executed after other sentence detection.

However, if  inputCols() for this annotator is an already sentence detection output, we will only be working on top of the first sentence. Thus, this needs to be fixed.

Reply:
https://github.com/JohnSnowLabs/spark-nlp/issues/407